### DISCUSSION TITLE
Notification Request: Renovate PR Creation/Update on Repo failure

### DISCUSSION THREAD

**Discussion by rohith1312** [renovatebot/renovate#25975] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.81.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### What is your question?

Hi,
Is there a way to achieve this? If Renovate skips creating or updating the pull request for a repository while running as a Docker image in a Jenkins job, we would like to receive a notification. Jenkins will notify us if the entire build fails, but say if we have 100 repo scans, then even if the PR creation or update fails for 1 repo due to some issues, Renovate will skip and continue for other repos without any notification about missed PR updates or creations. But if the job fails as per our configuration, it will notify us.
So we would like to know if there is any way that if the renovate skips PR creation or update for a repo then it should notify us.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Please be more specific about the exact error condition you are talking about, illustrate with logs

**Reply by rohith1312** [reply]
We have a scenario where we have two repositories. While Renovate successfully scans one repository, the other encounters issues(we 
wantedly created this situation) causing it to fail in creating or updating pull requests. Consequently, the Jenkins job concludes successfully. However, as one of the repositories either skipped or failed in pull request creation or updates, we would like to receive notifications indicating the repository name for which the PR creation or update failed.

please find the attachment for Jenkins logs
[renovatejenkinslogs.txt](https://github.com/renovatebot/renovate/files/13472621/renovatejenkinslogs.txt)

**Reply by rarkins** [reply]
There are no errors in that log, so you need to keep trying to be more specific.

Asking for alerts if a Repo doesn't get PRs is not a good idea. If there are specific types of failures you think should be surfaced, then be more specific.

**Reply by rohith1312** [reply]
I have tested a few more test cases and found the following logs:

Test Case 1:
I edited the main-major-dependencies branch with incorrect values, resulting in a failed attempt to create a pull request (PR). The relevant logs indicate:

{
  DEBUG: Renovate did not find a PR for branch #renovate/main-major-dependencies (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  DEBUG: branchExists=true (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  DEBUG: dependencyDashboardCheck=undefined (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  DEBUG: recreateClosed is true. No need to check for a closed PR. (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  DEBUG: Checking if PR has been edited (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  DEBUG: branch.isModified(): using git to calculate (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  DEBUG: branch.isModified() = true (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-major-dependencies)
  "branchName": "renovate/main-major-dependencies",
  "lastAuthor": "Rohith.Kamshetty@pega.com",
  "gitAuthorEmail": "****"
}

and

{
  "branchName": "renovate/main-major-dependencies",
  "prNo": null,
  "prTitle": "Update dependencies (main) (major)",
  "result": "pr-edited",
  "upgrades": {}
}

**Test Case 2:**
I edited the main-dependencies branch with incorrect values. An existing PR was open for it, but the attempt to update the PR failed. The logs reveal the following details:

{
DEBUG: findPr(renovate/main-dependencies, "undefined", "open") (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: getPrList() (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: returning cached PR list (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: Found PR #3 (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: getPr(3) (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: branchExists=true (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: dependencyDashboardCheck=undefined (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: PR rebase requested=false (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: Checking if PR has been edited (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: branch.isModified(): using git to calculate (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
DEBUG: branch.isModified() = true (repository=ALMADM/adaptive-learning-core-lib, baseBranch=main, branch=renovate/main-dependencies)
       "branchName": "renovate/main-dependencies",
       "lastAuthor": "Rohith.Kamshetty@pega.com",
       "gitAuthorEmail": "****"

}

and 

 {
           "branchName": "renovate/main-dependencies",
           "prNo": 3,
           "prTitle": "Update dependencies (main)",
           "result": "pr-edited",
           "upgrades": [
             }


No error or status code was reported for the above test case. It would be helpful to receive a notification in such scenarios.
[TestCase1logs.txt](https://github.com/renovatebot/renovate/files/13476091/TestCase1logs.txt)


[TestCase2logs.txt](https://github.com/renovatebot/renovate/files/13476094/TestCase2logs.txt)

**Reply by rarkins** [reply, marked as answer]
Sorry, but I don't think any of these cases are practical to raise errors for
